fibsem/milling/base.py

```diff
@@ -1,51 +1,54 @@
 from abc import ABC, abstractmethod
 from copy import deepcopy
-from dataclasses import dataclass, fields, field
-from typing import List, Union, Dict, Any, Tuple, Optional
+from dataclasses import dataclass, fields, field, asdict
+from typing import List, Union, Dict, Any, Tuple, Optional, Type, TypeVar, ClassVar, Generic
 
 from fibsem.microscope import FibsemMicroscope
 from fibsem.milling.config import MILLING_SPUTTER_RATE
 from fibsem.milling.patterning.patterns2 import BasePattern as BasePattern, get_pattern as get_pattern
 from fibsem.structures import FibsemMillingSettings, Point, MillingAlignment, ImageSettings, CrossSectionPattern
 
+
+TMillingStrategyConfig = TypeVar(
+    "TMillingStrategyConfig", bound="MillingStrategyConfig"
+)
+TMillingStrategy = TypeVar("TMillingStrategy", bound="MillingStrategy")
+
+
 @dataclass
 class MillingStrategyConfig(ABC):
     """Abstract base class for milling strategy configurations"""
-    
-    def to_dict(self):
-        return {}
+    _advanced_attributes: ClassVar[Tuple[str, ...]] = ()
+
+    def to_dict(self) -> Dict[str, Any]:
+        return asdict(self)
+
+    @classmethod
+    def from_dict(
+        cls: Type[TMillingStrategyConfig], d: Dict[str, Any]
+    ) -> TMillingStrategyConfig:
+        return cls(**d)
 
-    @staticmethod
-    def from_dict(d: dict) -> "MillingStrategyConfig":
-        return MillingStrategyConfig()
-    
-    @property
-    def required_attributes(self) -> Tuple[str]:
-        return [field.name for field in fields(self)]
-    
     @property
-    def advanced_attributes(self) -> List[str]:
-        if hasattr(self, "_advanced_attributes"):
-            return self._advanced_attributes
-        return []
+    def required_attributes(self) -> Tuple[str, ...]:
+        return tuple(f.name for f in fields(self))
```

Comment:
Missing advanced_attributes property
```suggestion
    def required_attributes(self) -> Tuple[str, ...]:
        return tuple(f.name for f in fields(self))
        
    @property
    def advanced_attributes(self) -> Tuple[str, ...]:
        """Attributes that are considered advanced and may not be required for all strategies."""
        return self._advanced_attributes
```

Reply:
Ah, good catch, thanks!